src/library_webgl.js

```diff
@@ -1785,7 +1785,7 @@ var LibraryGL = {
   glBufferSubData__sig: 'viiii',
   glBufferSubData: function(target, offset, size, data) {
 #if MAX_WEBGL_VERSION >= 2
-    if ({{{ isCurrentContextWebGL2() }}}) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
+    if ({{{ isCurrentContextWebGL2() }}} && size > 0) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
       GLctx.bufferSubData(target, offset, HEAPU8, data, size);
```

Comment:
This would lead to calling the WebGL 1 style function below if size <= 0.

How about the following?

```js
  glBufferSubData: function(target, offset, size, data) {
#if ASSERTIONS
  assert(size >= 0);
#endif

#if MAX_WEBGL_VERSION >= 2
     if ({{{ isCurrentContextWebGL2() }}}) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
       size && GLctx.bufferSubData(target, offset, HEAPU8, data, size);
       return;
     }
#endif
    GLctx.bufferSubData(target, offset, HEAPU8.subarray(data, data+size));
  },
```

Reply:
I was under the impression that bufferSubData could affect something in the buffer state, that's why I was letting it fall back to the webgl 1 style function, but since it doesn't affect anything... Might as well move the size check to also include webgl 1?

Reply:
> Might as well move the size check to also include webgl 1 to avoid the allocation?

That does make sense.. shortest code is probably to also do

```js
size && GLctx.bufferSubData(target, offset, HEAPU8.subarray(data, data+size));
```

Reply:
Hmm, actually, although generally it is preferable to "diligently" call the functions so that WebGL spec related validation and GL error state etc. is properly recorded, so in the absence of other functional difference, may be good to leave that out in the WebGL 1 path. That way the target and currently bound buffer etc. will be properly validated.